### DISCUSSION TITLE
Add ECMWF Open Data via API?

### DISCUSSION THREAD

**Discussion by guidocioni** [contrailcirrus/pycontrails#50] [Ideas]
**Description**
ECMWF-IFS low resolution data are available on the opendata server without the need of registration with the open analogue of ecmwf-api-client, https://github.com/ecmwf/ecmwf-opendata.
As both packages use a really similar architecture maybe it wouldn't be hard to implement this in the package?

**Comment by mlshapiro** [comment]
Hi @guidocioni - this does seem straightforward. It should be as simple as replacing the `ECMWFService` in `hres.py` with the `Client` in the `ecmwf-opendata` package.

Are you interested to use the open forecast data in your models?  Can you share any more about your use case?

**Comment by guidocioni** [comment]
Yeah, I was suggesting it because it is indeed straightforward.
The cool thing about IFS is that you can basically run a live contrails forecast, which you can't do with era5 data as it is delayed.
However, I realized that there are way less variables on the opendata server and that the resolution is quite coarse (0.4 deg) so you'll only be able to perform issr evaluation but no accf. It's up to you whether you want to include it or not :)

I'd suggest you take a look also at the DWD opendata server because that data is available for many variables at a much higher resolution so it would make a pretty good forecasting tool 😉

**Comment by mlshapiro** [comment]
Thanks! We are familiar with the DWD opendata server and hope to include more meteorology interfaces in time.

The `GFSForecast` allows access to forecast data without any credentials.
